### PR TITLE
Merge mit pdos [6] (related to #261)

### PR DESCRIPTION
**Pull request by kimjungwow**
### Rust 코드에 반영되었는지 한 번 더 확인 후 리뷰 요청 드리겠습니다 (related to #261 )

**2020/8/18**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/740d36373630e519117077f867437c9abbfd1f8d : Delete some obselete stuff
  - C 코드 수정 X -> **Rust 해당 사항 없음**

**2020/8/19**  
- [X] https://github.com/mit-pdos/xv6-riscv/commit/b33574df38f358ba2173caa92e8287b5b0cb9cff : Use the major number defined in file.h. The minor number is ignored;
  - `user/init.c` 수정 -> **Rust 해당 사항 없음**

**2020/8/20**  
- [X] https://github.com/mit-pdos/xv6-riscv/commit/aefa2697d705e316aa5255004e4b6a129e9afe2a : usertest for exec() out of memory recovery
  - `sysfile.c::sys_exec()`에서 `kalloc()==0`인 경우 panic이 아닌 -1을 리턴하도록 변경 -> **`sysfile.rs`에 적용 완료**
  - `exec.c::exec()`에서 `uvmalloc()`의 값을 `sz1` 변수에 잠시 넣는 방법 사용 -> **`exec.rs`에 동일하게 적용 완료**
  - `user/usertests.c` : execout usertest 추가 -> **Rust 해당 사항 없음**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/59a9863a17d881498bf7519878566f74341a1b71 : touch sbrk()-allocated memory to make sure it exists
    - `user/usertests.c` 수정 -> **Rust 해당 사항 없음**

**2020/8/21**    
- [X] https://github.com/mit-pdos/xv6-riscv/commit/efaa7b8e2ae2ae059f3985bda718d91f3501b294 : Adjust a few prototypes to be explicit that they don't take arguments
  - `defs.h` : 특정 함수들이 인자를 받지 않는다는 것을 명확히 하기 위해 `void kinit()`을 `void kinit(void)` 등으로 변경 -> **Rust 해당 사항 없음**

**2020/8/22**      
- [X] https://github.com/mit-pdos/xv6-riscv/commit/3c25112f382e6ccdfda927125cfe30d2b7d6b9df : Delete some obselete stuff
  - merge commit -> **이미 반영 완료**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/d4d487731ad760376999130fc2ad520c3dd6166c : Use the major number defined in file.h. The minor number is ignored;
    - merge commit -> **이미 반영 완료**  
- [X] https://github.com/mit-pdos/xv6-riscv/commit/5860dcd07d9a9cba5c7ebf7488636d131015316d : usertest for exec() out of memory recovery
    - merge commit -> **이미 반영 완료**  
- [X] https://github.com/mit-pdos/xv6-riscv/commit/2a4a8764a52cccc128aa1a6d7b321bafc14b48e6 : touch sbrk()-allocated memory to make sure it exists
    - merge commit -> **이미 반영 완료**  
- [X] https://github.com/mit-pdos/xv6-riscv/commit/0bb2533f73fb34348072da49b7970b08bfd9ac27 : Adjust a few prototypes to be explicit that they don't take arguments
    - merge commit -> **이미 반영 완료**  

**2020/8/27**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/d7e5f269106f4b76fcbd90331a3b44f879ee9c54 : fix usertests to pass all the riscv-sol-fall20 solutions.
  - `user/usertests.c` 수정 -> **Rust 해당 사항 없음**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/2ec9c6ed66629fd5b833f06af2080eab46c0af35 : Remove obselete ifdef that shouldn't have been there ever
  - `user/usertests.c` 수정 -> **Rust 해당 사항 없음**

**2020/8/28**  
- [X] https://github.com/mit-pdos/xv6-riscv/commit/db067d24da0b6472afe9ce4165e0868772b11964 : suppress write() output after panic()
  - `console.c::consolewrite()`에서 `cons.lock` acquire 시도하기 전에 `panicked!=0`이면 spin함 -> **`console.rs::consolewrite`에서 `kernel().is_panicked()`확인 후 `spin_loop();`하도록 변경 완료**
  - `printf.c::panic()` 내 주석 변경 -> **Rust 해당 사항 없음**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/ffb2ee074a42ed30a87fd6804682fa62eca13500 : move panicked check to uart.c
  - `panicked!=0`이면 spin하는 코드를 `console.c`에서 `uart.c`로 이동 -> xv6의 의도를 살려 **`console.rs`의 spin하는 코드를 `uart.rs::putc(), putc_sync()`으로 이동 완료**
  - `printf.c::panic()` 내 주석 변경 -> **Rust 해당 사항 없음**
  - `user/grind.c` -> **Rust 해당 사항 없음**
- [X] https://github.com/mit-pdos/xv6-riscv/commit/0b4e78f3a94baae548b535ac8de80d2ce4c963c8 : fix usertests to pass all the riscv-sol-fall20 solutions.
- merge commit -> **이미 반영 완료** 
- [X] https://github.com/mit-pdos/xv6-riscv/commit/de434fd933a07fba80d16441a5cd70ebebcde518 : Remove obselete ifdef that shouldn't have been there ever
- merge commit -> **이미 반영 완료** 
- [X] https://github.com/mit-pdos/xv6-riscv/commit/748809e1b36cd08b21ddfdf79486787cc5c85453 : suppress write() output after panic()
- merge commit -> **이미 반영 완료** 
- [X] https://github.com/mit-pdos/xv6-riscv/commit/490f91ab53e22d90afb1ee37d5148e341ccf538e : move panicked check to uart.c
- merge commit -> **이미 반영 완료**

### CODE REVIEW

**Comment by kimjungwow** [comment]
@jeehoonkang 교수님, 리뷰 요청드립니다. 2020/8/18 ~ 2020/8/28 의 커밋들입니다.

File: kernel-rs/src/sysfile.rs [hunk 1]
```diff
@@ -332,7 +332,10 @@ impl Kernel {
                 break;
             }
 
-            *arg = self.alloc().expect("sys_exec kalloc").into_usize() as *mut _;
+            match self.alloc() {
```
Comment (jeehoonkang) [inline kernel-rs/src/sysfile.rs:335]: `some_or!(..., break);` 으로 합시다.

**Review by jeehoonkang — CHANGES_REQUESTED** [review]
사소한거 수정하고 머지할게요

File: kernel-rs/src/sysfile.rs (hunk 1 above)
Comment (kimjungwow) [reply kernel-rs/src/sysfile.rs]: https://github.com/kaist-cp/rv6/pull/296/commits/6cbfcb80992bdda699db750eb90fe2d695eafb7b 반영하였습니다 :)

**Pull request merged**